### PR TITLE
User profile updates

### PR DESCRIPTION
**Pull request by gdestuynder**
This takes into account all recent comments/feedback through different channels.
Please SQUASH this PR when merging. I left separate comments for clarity in case walking through updates is needed, but this is too much noise for the repo otherwise.

Summary of fixes:

- Added ACLs
- Added OIDC attributes
- Standardized attribute and acl lists to always be lists with structs that have a name (as agreed)
- Added some missing fields

I would propose that we use this as the base profile to implement CIS (MINUS the OIDC tags that are only here for informational reasons, since these will be added by auth0 automagically).
As we find potential needs, shortcomings, or user stories are added we can extend or modify the profile to adjust for the CIS. When CIS goes in prod that'll be it for this profile version.

We can add version support later, if a new version appears - though we can't really expect RP's will update code to read new versions, it will only give them an option to read a newer schema if we'd ever need that.

### CODE REVIEW

**Review by andrewkrug — COMMENTED** [review]
This looks straight forward to me.  I like the multi-valued lists.

File: doc/User_profile.json [hunk 1]
```diff
@@ -1,43 +1,145 @@
...
-				"ssh": "4096 SHA256:rlhYcDi2e98sekBwu9aaPEmr3xsg1CsS1tGXksSTSIg test (RSA)",
-				"pgp": "0x61AB87D6F66CE2FCD2D2E1F56A65DFA844722517"
-			},
-			"picture": "https://s.gravatar.com/avatar/...png",
-			"shirtSize": "L"
+	"userName": "jdoe@mozilla.com",
+	"displayName": "john doe",
+	"firstName": "john",
+	"lastName": "doe",
+	"preferredLanguage": "en_US",
```
Comment (akatsoulas) [inline doc/User_profile.json:22]: Should we also keep a list of languages that a user can/prefers to communicate similar to the `emails`?

**Review by johngian — APPROVED** [review]

**Comment by andrewkrug** [comment]
Merged to move forward.

File: doc/User_profile.json (hunk 1 above)
Comment (gdestuynder) [reply doc/User_profile.json:22]: we can do that

**Pull request merged**